docs/how_to_contribute/math/how_to_add_prime_fields.md

```diff
@@ -0,0 +1,79 @@
+# How to add prime fields
+
+Follow this guide to add a new prime field for Tachyon.
+
+_Note_: We have our own development conventions. Please read the [conventions docs](/docs/how_to_contribute/conventions.md) before contributing.
+
+## 1. Add a `BUILD.bazel` file
+
+Begin by creating a directory named `<new_prime_field>_prime` in `/tachyon/math/finite_field/`. Add a `BUILD.bazel` into this directory. Note that once parameters are added to `BUILD.bazel`, Bazel will automatically generate the prime field code based on these parameters when it builds the target.
+
+Choose among `generate_prime_fields()`, `generate_fft_prime_fields()`, or `generate_large_fft_prime_fields()` for code generation, depending on the prime field type. For more information, refer to the [`prime_field_generator`](/tachyon/math/finite_fields/generator/prime_field_generator/build_defs.bzl).
+
+For instance, to implement a FFT prime field, create a directory (`/tachyon/math/finite_field/<new_prime_field>`) and add a `BUILD.bazel` file as shown below:
+
+```bazel
+# /tachyon/math/finite_field/<new_prime_field>/BUILD.bazel
+
+load("@bazel_skylib//rules:common_settings.bzl", "string_flag")
+load("//bazel:tachyon_cc.bzl", "tachyon_cc_library")
+load(
+    "//tachyon/math/finite_fields/generator/prime_field_generator:build_defs.bzl",
+    "SUBGROUP_GENERATOR",
+    "generate_fft_prime_fields", # NOTE: Choose generator type
+)
+
+package(default_visibility = ["//visibility:public"])
+
+string_flag(
+    name = "fr_" + SUBGROUP_GENERATOR,
+    build_setting_default = "{subgroup_generator}", # input Subgroup generator value
+)
+
+generate_fft_prime_fields( # NOTE: Choose generator type
+    name = "new_prime_field",
+    class_name = "NewPrimeField",
+    modulus = "{moulus_value}", # input modulus value
+    namespace = "tachyon::math",
+    subgroup_generator = ":fr_" + SUBGROUP_GENERATOR,
+)
+
+tachyon_cc_library(
+    name = "prime_field_<new_prime_field>", 
+    hdrs = ["prime_field_<new_prime_field>.h"],
+    defines = ["TACHYON_POLYGON_ZKEVM_BACKEND"],
+    deps = [
+        "//tachyon/math/base/gmp:gmp_util",
+        "//tachyon/math/finite_fields:prime_field_base",
+    ],
+)
+```
+
+_Note_: Adjust the file contents based on the desired specifications.
```

Comment:
Instead of this line, just put examples of similar files with a similar line to what u have later... aka
"Use the following files for reference:"
- ....